src/scriptdev2/scripts/kalimdor/the_barrens.cpp

```diff
@@ -591,6 +592,123 @@ CreatureAI* GetAI_npc_wizzlecranks_shredder(Creature* pCreature)
+## Quest Counterattack!
+QuestID 4,021
+#####*/
+enum
+{
+
+	NPC_KROMZAR = 9456,
+	NPC_KOLKAR_INVADER = 9524,
+	NPC_KOLKAR_STORMSEER = 9523,
+	NPC_HORDE_DEFENDER = 9457,
+	NPC_HORDE_AXE_THROWER = 9458,
+	NPC_REGTHAR_DEATHGATE = 3389,
+	QUEST_COUNTERATTACK = 4021,
+
+};
+
+struct npc_regthar_deathgateAI : public ScriptedAI {
+	npc_regthar_deathgateAI(Creature* pCreature) : ScriptedAI(pCreature) { Reset(); }
+
+
+
+
+	uint8 m_uiMobCount;
+
+	ObjectGuid m_playerGuid;
+
+	void Reset() override
+	{
+		m_playerGuid.Clear();
+	}
+
+	void JustDied(Unit* /*pKiller*/) override
+	{
+		Player* pPlayer = m_creature->GetMap()->GetPlayer(m_playerGuid);
+
+		if (pPlayer && pPlayer->GetQuestStatus(QUEST_COUNTERATTACK) == QUEST_STATUS_INCOMPLETE)
+			pPlayer->SendQuestFailed(QUEST_COUNTERATTACK);
+	}
+
+	void JustSummoned(Creature* pSummoned) override
+	{
+		pSummoned->AI()->AttackStart(m_creature);
+
+	}
+
+
+	void SummonedCreatureJustDied(Creature* /*pKilled*/) override
+	{
+		++m_uiMobCount;
+		if (m_uiMobCount == 10)
+			m_creature->SummonCreature(NPC_KROMZAR, -298.0f, -1836.0f, 94.06f, 0.0f, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 900000); ///Summon in the middle of the invisible box
+
+		if (m_uiMobCount % 4 == 0) //for every 4 units kiled, bring back up
+			m_creature->SummonCreature(NPC_HORDE_DEFENDER, -290.512054f, -1872.060547f, 92.658226f, 0.0f, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 600000);
+		if (m_uiMobCount)
+			return;
+
+
+		Player* pPlayer = m_creature->GetMap()->GetPlayer(m_playerGuid);
+
+		if (pPlayer && pPlayer->GetQuestStatus(QUEST_COUNTERATTACK) == QUEST_STATUS_INCOMPLETE)
+			pPlayer->GroupEventHappens(QUEST_COUNTERATTACK, m_creature);
+
+		m_playerGuid.Clear();
+		m_creature->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_QUESTGIVER);
+	}
+
+
+	void StartEvent(Player* pPlayer)
+	{
+		srand(time(NULL));
+		//creates a box where all the centaurs could spawn from
+		float xLO = -347.081238f;
+		float xHI = -260.881226f;
+		float yLO = -1874.357178f;
+		float yHI = -1805.872681f;
+		float zLO = 91.948669f;
+		float zHI = 94.825653f;
+		//------------------------------------------
+		///Spawn centaurs
+		for (int i = 0; i < 25; i++) {
+
+			int c = i % 2;
+			float ori = static_cast <float> (rand()) / (static_cast <float> (RAND_MAX / (4)));
+			float xcoord = xLO + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX / (xHI - xLO))); //prints random number between LO and HI
+			float ycoord = yLO + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX / (yHI - yLO)));
+			float zcoord = zLO + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX / (zHI - zLO)));
+			m_creature->SummonCreature(9523 + c, xcoord, ycoord, zcoord, ori, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 900000);
+			//-------------------------------------------
+		}
+
+		///Summons helpers, --Could be more effienct maybe.
+		m_creature->SummonCreature(NPC_HORDE_DEFENDER, -294.567017f, -1873.857300f, 92.453888f, 0.0f, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 900000);
```

Comment:
Found this for starters;

Id: 13965
Name: Create Krom'zar's Banner
General info
Category = 0, SpellIconId = 289, ActiveIconId = 0, SpellVisual = (0, 0)
SpellFamilyName = SPELLFAMILY_GENERAL, SpellFamilyFlags = 0x0000000000000000
SpellSchoolMask = 1 (SCHOOL_MASK_PHYSICAL, SCHOOL_MASK_ALL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
Advanced info
Spell Level = 1, BaseLevel 0, MaxLevel 0, MaxTargetLevel 0
Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange (Id 2) "Combat Range"
MinRange = 0
MaxRange = 5
CastingTime (Id 1) = 0.00
Duration: ID (5) 300000, 0, 300000
Interrupt Flags: 0x00000000, AuraIF 0x00000080, ChannelIF 0x00000000
Chance = 101, charges - 0
Effects
Effect 0:
Id: 76 (SPELL_EFFECT_SUMMON_OBJECT_WILD)
BasePoints = 1
Targets (18, 0) (TARGET_EFFECT_SELECT, NO_TARGET)
EffectMiscValueA = 164690
Effect 1:
NO EFFECT
Effect 2:
NO EFFECT